scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -43,6 +46,36 @@
             :reject-after-match-attempt true})
          node-name->available)))
 
+(defn states-to-scan
+  "Determine all states to scan by unioning task id's from expected and existing state maps. "
+  [{:keys [expected-state-map existing-state-map] :as kcc}]
+  (->>
+    (set/union (keys @expected-state-map) (keys @existing-state-map))
+    (into #{})))
+
+(defn scan-states
+  "Scan all states. Note: May block or be slow due to rate limits."
+  [kcc]
+  (log/info "Starting state scan: " )
+  ; TODO Add in rate limits; only visit non-running/running states so fast.
+  ; TODO Add in maximum-visit frequency. Only visit a state once every XX seconds.
+  (let [states-to-scan (states-to-scan kcc)
+        _ (log/info "Doing state scan. Visiting" (count states-to-scan) "states")]
+    (doseq [state states-to-scan]
```

Comment:
Is `state` a task ID? If so, can we rename these variables to make it clear?

Reply:
Yup.